platform/darwin/src/MGLShapeSource.h

```diff
@@ -41,6 +41,24 @@ FOUNDATION_EXTERN MGL_EXPORT const MGLShapeSourceOption MGLShapeSourceOptionClus
  */
 FOUNDATION_EXTERN MGL_EXPORT const MGLShapeSourceOption MGLShapeSourceOptionClusterRadius;
 
+/**
+ An `NSDictionary` object where the key is an `NSString`. The dictionary key will be the attribute feature attribute key. The resulting attribute value is aggregated from the clustered points.
```

Comment:
This paragraph seems redundant. It's clear if starts with:

> The dictionary key is an `NSString` that will be an attribute key for the clusters. The dictionary value is an `NSArray` consisting of a reduce operator and a map expression.

Reply:
Another question: Why the array also holds a string? shouldn't hold expression values only?

Reply:
This is the same behavior ported from GL-JS. For reduceExpression, either a string with a operand key word or a valid expression containing  `featureAccumulated` are acceptable.

Reply:
I see. In that case let's drop the term `reduce` as it is foreign to iOS expressions. The array should hold a `sum`/`max` function instead of an `NSString` or in opposite side a constant `NSString` expression. I would prefer the first option as this code is parsed back as an `NSExpression` in https://github.com/mapbox/mapbox-gl-native/pull/15515/files#diff-782f4462e342b786fea8a94199fb6225R115-R116

cc @zmiao @jmkiley

Reply:
@fabian-guerra, do you mean what @jmkiley mentioned here: https://github.com/mapbox/mapbox-gl-native/pull/15515#issuecomment-539266132

Reply:
> In that case let's drop the term reduce as it is foreign to iOS expressions.

Have we done this yet? I appreciate the benefit that changing a name might provide to platform-specific developers, but I'd like to present an alternate perspective: making decisions such as these makes our features harder to use. 

If I'm understanding correctly, `reduceExpression` is not a term of art: it's something that @mourner created and named for clusterProperties when designing it in GL-JS. Is there a different name for this expression that would allow an iOS developer to immediately grok what the function does? If so, let's consider it. If not, then that iOS developer will still have to rely on Mapbox documentation to learn about the expression, and we should reduce this friction as much as possible.

The documentation available isn't limited to a single platform's API reference. Ways for people to learn about Mapbox features include technical guides, examples, blog posts. It's unrealistic to adapt every resource we make to every platform, and I have a strong opinion that the more consistently we name our features across all platforms, the better of an experience our major customers will have building maps with Mapbox and porting features into all of the platforms they need to support.